cli/crates/cli/src/logs.rs

```diff
@@ -0,0 +1,167 @@
+use std::str::FromStr;
+
+use backend::api::{
+    errors::ApiError,
+    logs::{self as api, LogEventsRange},
+};
+use common::types::{LogLevel, UdfKind};
+use itertools::Itertools;
+
+use crate::errors::CliError;
+
+#[derive(Debug)]
+pub struct BranchReference {
+    account_slug: String,
+    project_slug: String,
+    branch_name: Option<String>,
+}
+
+pub async fn project_branch_reference_to_account_project_slug(
+    project_reference: String,
+) -> Result<BranchReference, CliError> {
+    let elements = project_reference.split('/').collect_vec();
+    Ok(match elements.as_slice() {
+        [account_slug, project_slug] => BranchReference {
+            account_slug: account_slug.to_string(),
+            project_slug: project_slug.to_string(),
+            branch_name: None,
+        },
+        [project_slug] if !project_slug.contains('.') => {
+            let account_slug = api::personal_account_slug().await.map_err(CliError::BackendApiError)?;
+            BranchReference {
+                account_slug,
+                project_slug: project_slug.to_string(),
+                branch_name: None,
+            }
+        }
+        _ => {
+            let domain = url::Url::from_str(&project_reference)
+                .ok()
+                .and_then(|parsed_url| parsed_url.domain().map(str::to_owned))
+                .unwrap_or(project_reference.clone());
```

Comment:
`.unwrap_or_else(|| project_reference.clone())` would make sense to avoid cloning on the happy path. Although here it's almost certainly premature optimization.

Reply:
Thanks, good catch!